### PR TITLE
Get total number of days in a BS year

### PR DESCRIPTION
**Pull request by mesaugat**
* Added a function to retrieve total number of days in a particular BS year
* Added tests for the new function

This is a nice addition to the `dateconv` package even though it isn't really needed for AD to BS conversion.

Another motive for this PR is that I need this in something that I'm working on. 😆

### CODE REVIEW

File: dateconv/dateconv.go [hunk 1]
```diff
@@ -72,6 +73,22 @@ func BsDaysInMonthsByYear(yy int, mm time.Month) (int, bool) {
 	return months[query], true
 }
 
+// TotalDaysInBSYear returns total number of days in a particular BS year.
+func TotalDaysInBSYear(bsYear int) (int, error) {
+	sum := 0
+	days, ok := bsDaysInMonthsByYear[bsYear]
+
+	if ok == false {
```
Comment (srishanbhattarai) [inline dateconv/dateconv.go]:
```suggestion
	if !ok {
```

File: dateconv/dateconv.go [hunk 2]
```diff
@@ -72,6 +73,22 @@ func BsDaysInMonthsByYear(yy int, mm time.Month) (int, bool) {
 	return months[query], true
 }
 
+// TotalDaysInBSYear returns total number of days in a particular BS year.
+func TotalDaysInBSYear(bsYear int) (int, error) {
+	sum := 0
```
Comment (srishanbhattarai) [inline dateconv/dateconv.go]: Please initialize this right above the `for` loop. This feels very C-like to me 😄

File: dateconv/dateconv.go [hunk 3]
```diff
@@ -72,6 +73,22 @@ func BsDaysInMonthsByYear(yy int, mm time.Month) (int, bool) {
...
+	days, ok := bsDaysInMonthsByYear[bsYear]
+
+	if ok == false {
+		return -1, fmt.Errorf("Year should be in between %d and %d", bsLBound, bsUBound)
+	}
+
+	for _, value := range days {
+		sum = sum + value
```
Comment (srishanbhattarai) [inline dateconv/dateconv.go]:
```suggestion
		sum += value
```

**Comment by mesaugat** [comment]
@srishanbhattarai I thought of adding it to the test struct as well but eventually went against it.

**Pull request merged**